src/sas/qtgui/Perspectives/Fitting/UnitTesting/ConstraintWidgetTest.py

```diff
@@ -82,38 +84,233 @@ def testOnFitTypeChange(self):
+        self.widget.currentType = "Batch"
         self.assertFalse(self.widget.isTabImportable("BatchTab"))
+        self.widget.currentType = "test"
+        self.assertFalse(self.widget.isTabImportable("test_tab"))
+        test_tab.data_is_loaded = True
+        self.assertTrue(self.widget.isTabImportable("test_tab"))
 
     def testOnTabCellEdit(self):
         ''' test what happens on monicker edit '''
-        # Mock the datafromitem() call from FittingWidget
-        data = Data1D(x=[1,2], y=[1,2])
-        GuiUtils.dataFromItem = MagicMock(return_value=data)
-        item = QtGui.QStandardItem("test")
-        self.perspective.addFit([item])
+        # Mock a tab
+        test_tab = MagicMock(spec=FittingWidget)
+        test_tab.data_is_loaded = False
+        test_tab.kernel_module = MagicMock()
+        ObjectLibrary.getObject = MagicMock(return_value=test_tab)
+        self.widget.updateFitLine("test_tab")
+
+        # disable the tab
+        self.widget.tblTabList.item(0, 0).setCheckState(0)
+        self.assertEqual(self.widget.tabs_for_fitting["test_tab"], False)
+        self.assertFalse(self.widget.cmdFit.isEnabled())
+        # enable the tab
+        self.widget.tblTabList.item(0, 0).setCheckState(2)
+        self.assertEqual(self.widget.tabs_for_fitting["test_tab"], True)
+        self.assertTrue(self.widget.cmdFit.isEnabled())
 
     def testUpdateFitLine(self):
         ''' See if the fit table row can be updated '''
-        pass
+        # mock a tab
+        test_tab = MagicMock(spec=FittingWidget)
+        test_tab.data_is_loaded = False
+        test_tab.kernel_module = MagicMock()
+        test_tab.kernel_module.name = "M1"
+        ObjectLibrary.getObject = MagicMock(return_value=test_tab)
+
+        # Add a tab without an constraint
+        self.widget.updateFitLine("test_tab")
+        self.assertEqual(self.widget.tblTabList.rowCount(), 1)
+        # Constraint tab should be empty
+        self.assertEqual(self.widget.tblConstraints.rowCount(), 0)
+
+        # Add a second tab with an active constraint
+        test_tab.getComplexConstraintsForModel = MagicMock(
+            return_value=[('scale', self.constraint1.func)])
+        test_tab.getFullConstraintNameListForModel = MagicMock(
+            return_value=[('scale', self.constraint1.func)])
+        test_tab.getConstraintObjectsForModel = MagicMock(
+            return_value=[self.constraint1])
+        self.widget.updateFitLine("test_tab")
+        # We should have 2 tabs in the model tab
+        self.assertEqual(self.widget.tblTabList.rowCount(), 2)
+        # One constraint in the constraint tab
+        self.assertEqual(self.widget.tblConstraints.rowCount(), 1)
+        # Constraint should be active
+        self.assertEqual(self.widget.tblConstraints.item(0, 0).checkState(), 2)
+        # Check the text
+        self.assertEqual(self.widget.tblConstraints.item(0, 0).text(),
+                         test_tab.kernel_module.name +
```

Comment:
PEP8: prefer operator on the start of the next line... this makes it easier to see that it is not an additional function argument.